### PR TITLE
#1661 Delete skip and xfail tests related to route removal

### PR DESCRIPTION
**Pull request by MackHalliday**
# Description

- Deleting skip and xfail tests related to route removal. 
- Remove any import statement no longer required related to deleted tests
- [See Google Sheets on all files impacted](https://docs.google.com/spreadsheets/d/1xiQwDi-DRTws3S6k2XPyXFOTIaVNzFHI-Vb7fVLQvGg/edit#gid=0&fvid=1213040269)

issue #1661 

## Type of change

Please check the relevant option(s).

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

- Re-running test suite locally 
- [Deployed to DEV thru Github Actions](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/8071031645)

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] (N/A) I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] (N/A) I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [ ] (N/A) Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

File: tests/app/service/test_rest.py [hunk 1]
```diff
@@ -2009,7 +1248,9 @@ def test_get_services_with_detailed_flag_defaults_to_today(client, mocker):
     assert resp.status_code == 200
 
 
-@pytest.mark.skip(reason='Do we utilize this?')
+@pytest.mark.skip(
+    reason='Used by GET /service if arg detailed TRUE- keeping route for v3, unsure if keep optional arg detailed.'
```
[the remark this answers is not in the document]
Comment (MackHalliday) [reply tests/app/service/test_rest.py:1252]: [See file and method here](https://github.com/department-of-veterans-affairs/notification-api/blob/master/app/service/rest.py#L122)

**Comment by MackHalliday** [comment]
@kalbfled - Performed self-check on code and deployed to dev. 

PR still in draft and I have not moved our user story to Dev Testing.

**Comment by MackHalliday** [comment]
@kalbfled - [Test accidentally removed](https://github.com/department-of-veterans-affairs/notification-api/pull/1666/commits/1828d9fda46377fbab43007516bbfaaa90dedf16). 

Reverted  commit 1828d9fda46377fbab43007516bbfaaa90dedf16

Reran ruff.

File: tests/app/service/test_rest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalbfled) [reply tests/app/service/test_rest.py:1252]: We decided not to remove these.

**Comment by EvanParish** [comment]
Please change the label on the PR to one listed [here](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml)

File: tests/app/celery/test_letters_pdf_tasks.py
Comment (EvanParish) [inline tests/app/celery/test_letters_pdf_tasks.py:1]: Is there a reason to keep this file around? Most of the tests where removed and the rest all relate to functionality we don't use. Thoughts @k-macmillan?

**Comment by kalbfled** [comment]
> Please change the label on the PR to one listed [here](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml)

I added back the "bug" label, but that doesn't reflect what this is.

File: tests/app/letters/test_letter_utils.py [hunk 2]
```diff
@@ -46,38 +43,6 @@ def _sample_precompiled_letter_notification_using_test_key(sample_precompiled_le
...
 
-@pytest.mark.parametrize(
-    'created_at,folder',
-    [
-        (datetime(2017, 1, 1, 17, 29), '2017-01-01'),
-        (datetime(2017, 1, 1, 17, 31), '2017-01-02'),
-    ],
-)
-@pytest.mark.skip(reason='Letter feature')
-def test_get_bucket_name_and_prefix_for_notification_valid_notification(
```
Comment (k-macmillan) [inline tests/app/letters/test_letter_utils.py:57]: This does not appear to just be a letter feature. Two things are being tested here, one happens to be a letter bucket.

File: tests/app/notifications/test_process_notifications.py [hunk 3]
```diff
@@ -787,37 +787,6 @@ def test_persist_sms_notification_stores_normalised_number(
         notify_db_session.session.commit()
 
 
-@pytest.mark.skip(reason='Endpoint slated for removal. Test not updated.')
```
Comment (k-macmillan) [inline tests/app/notifications/test_process_notifications.py:790]: This is wrong. This is not slated for removal.

File: tests/app/notifications/test_process_notifications.py [hunk 4]
```diff
@@ -866,35 +835,6 @@ def test_persist_letter_notification_finds_correct_postage(
...
```
Comment (k-macmillan) [inline tests/app/notifications/test_process_notifications.py:869]: This is wrong. Not slated for removal. This is one of those cases I warned about, they used `LETTER_TYPE` but the test is not letter-specific.

File: tests/app/celery/test_reporting_tasks.py [hunk 5]
```diff
@@ -469,56 +469,6 @@ def test_create_nightly_billing_for_day_use_BST(
...
-        sent_by=None,
-        international=False,
-        rate_multiplier=1.0,
-        billable_units=1,
-    )
-
-    stmt = select(FactBilling).where(FactBilling.template_id == template.id)
-    assert (notify_db_session.session.scalars(stmt).all()) == 0
-
-    create_nightly_billing_for_day('2018-01-14')
```
Comment (EvanParish) [inline tests/app/celery/test_reporting_tasks.py:496]: This function is still in use, why was this test marked as "not in use"?

**Review by EvanParish — CHANGES_REQUESTED** [review]

**Comment by kalbfled** [comment]
@MackHalliday is creating a new PR.

File: tests/app/letters/test_letter_utils.py (hunk 2 above)
Comment (MackHalliday) [reply tests/app/letters/test_letter_utils.py:57]: @k-macmillan This test is testing the method `get_bucket_name_and_prefix_for_notification` which is only used by which is only used by the method `get_letter_pdf` which is only used by the route `/service/<uuid:service_id>/template/preview/<uuid:notification_id>/<file_type>`

Route `/service/<uuid:service_id>/template/preview/<uuid:notification_id>/<file_type>` is slated for route removal according to our [ VA Notify Endpoints Audit ](https://docs.google.com/spreadsheets/d/18AfqlEOWlrnigauO57jA0VP3RMHTaP4luknqBxrcbuY/edit#gid=115873237)doc

Should this test therefore be removed?

Comment (MackHalliday) [reply tests/app/letters/test_letter_utils.py:57]: This is also a test under for Letter Utilities

Comment (k-macmillan) [reply tests/app/letters/test_letter_utils.py:57]: Yes, should be removed.

**Pull request closed without merging**